### PR TITLE
introduce the ergoDebug

### PR DESCRIPTION
**Pull request by leonlee2013**
Print out goroutine id, line number, file name, function name for better code debugging.

### CODE REVIEW

**Comment by halturin** [comment]
May I ask you to provide an example of debug output?

**Comment by leonlee2013** [comment]
**Example of debugging output:**
```bash
$ ./go_node1  -ergo.trace  

to stop press Ctrl-C
2022/12/05 18:45:26 [go_node1@127.0.0.1] CORE registering process: <3FA645AA.0.1001>
2022/12/05 18:45:26 [go_node1@127.0.0.1] CORE spawn a new process <3FA645AA.0.1001> (registered name: "")
2022/12/05 18:45:26 [go_node1@127.0.0.1] CORE registering name (<3FA645AA.0.1002>): system_app_sup
2022/12/05 18:45:26 [go_node1@127.0.0.1] CORE registering process: <3FA645AA.0.1002>
2022/12/05 18:45:26 [go_node1@127.0.0.1] CORE spawn a new process <3FA645AA.0.1002> (registered name: "system_app_sup")
2022/12/05 18:45:26 [go_node1@127.0.0.1] SUPERVISOR "system_app_sup" with restart strategy: one_for_one[permanent] 
2022/12/05 18:45:26 [go_node1@127.0.0.1] CORE registering name (<3FA645AA.0.1003>): system_metrics

```

```bash
$ ./go_node1 -ergo.debug  -ergo.trace 
2022/12/05 18:46:01 <Go#1>@core.go:402 node.(*core).newProcess [go_node1@127.0.0.1] CORE registering process: <3FA645AA.0.1002>
2022/12/05 18:46:01 <Go#1>@core.go:450 node.(*core).spawn [go_node1@127.0.0.1] CORE spawn a new process <3FA645AA.0.1002> (registered name: "system_app_sup")
2022/12/05 18:46:01 <Go#1>@supervisor.go:120 gen.(*Supervisor).ProcessInit [go_node1@127.0.0.1] SUPERVISOR "system_app_sup" with restart strategy: one_for_one[permanent] 
2022/12/05 18:46:01 <Go#22>@core.go:391 node.(*core).newProcess [go_node1@127.0.0.1] CORE registering name (<3FA645AA.0.1003>): system_metrics
2022/12/05 18:46:01 <Go#22>@core.go:402 node.(*core).newProcess [go_node1@127.0.0.1] CORE registering process: <3FA645AA.0.1003>
2022/12/05 18:46:01 <Go#22>@core.go:450 node.(*core).spawn [go_node1@127.0.0.1] CORE spawn a new process <3FA645AA.0.1003> (registered name: "system_metrics")

```

File: lib/tools.go [hunk 1]
```diff
@@ -50,19 +55,40 @@ func init() {
 	flag.BoolVar(&ergoTrace, "ergo.trace", false, "enable/disable extended debug info")
```
Comment (halturin) [inline lib/tools.go:55]: please, update this desc to "enable/disable extended node logging info"

Comment (halturin) [reply lib/tools.go:55]: Everything else looks good.

Comment (leonlee2013) [reply lib/tools.go]: ok

**Review by halturin — APPROVED** [review]

**Review by leonlee2013 — COMMENTED** [review]
Does the code still need to be modified?

**Comment by leonlee2013** [comment]
Can you merge  this pull request?
I need to delete my repository and refork ergo-service to easily pull a new branch from v221 for other PR.

**Review by halturin — APPROVED** [review]

**Pull request merged**